includes/class-omise-callback.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * @since 4.0
+ */
+class Omise_Callback {
+	/**
+	 * @var \Omise_Callback
+	 */
+	protected static $the_instance = null;
+
+	/**
+	 * @static
+	 *
+	 * @return \Omise_Callback - The instance.
+	 */
+	public static function instance() {
+		if ( is_null( self::$the_instance ) ) {
+			self::$the_instance = new self();
+		}
+
+		return self::$the_instance;
+	}
+
+	public static function execute() {
+		$callback = self::instance();
+
+		$order = $callback->retrieve_order();
+		$order->add_order_note( __( 'OMISE: Validating the payment result...', 'omise' ) );
+
+		try {
+			$charge = OmiseCharge::retrieve( $order->get_transaction_id() );
+
+			switch ( strtolower( $charge['status'] ) ) {
```

Comment:
There are two more potential charge statuses: `expired` and `reversed` https://www.omise.co/charges-api